overlord/ifacestate/ifacestate.go

```diff
@@ -101,7 +101,8 @@ func connect(st *state.State, plugSnap, plugName, slotSnap, slotName string, fla
 	//  - connect task
 	//  - connect-slot-<slot> hook
 	//  - connect-plug-<plug> hook
-	// The tasks run in sequence (are serialized by WaitFor).
+	// The tasks run in sequence (are serialized by WaitFor). The hooks are optional
+	// and their tasks are not created if not provided by snap.
```

Comment:
A positive scenario may be easier to follow than a double-negative one:
`... tasks are created when hook exists or is declared in the snap`